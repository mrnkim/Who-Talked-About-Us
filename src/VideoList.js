import React from "react";
import { Container, Row, Col, Button, Form } from "react-bootstrap";
import Video from "./Video";
import { v4 as uuidv4 } from "uuid";

function VideoList({ index_id, videos, deleteVideo }) {
  return videos.data.map((data) => (
    <Col sm={12} md={6} lg={4} xl={3} className="mb-4" key={uuidv4()}>
      <Video
        index_id={index_id}
        video_id={data._id}
        deleteVideo={deleteVideo}
        showDeleteButton={true}
        key={uuidv4()}
      />
    </Col>
  ));
}

export default VideoList;
