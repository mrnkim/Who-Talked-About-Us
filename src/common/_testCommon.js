const TEST_INDEXES = {
  data: [
    { _id: 1, index_name: "INDEX1" },
    { _id: 2, index_name: "INDEX2" },
  ],
  isLoading: false,
};

const TEST_VIDEOS = {
  data: [
    { _id: 1, video_name: "VIDEO1" },
    { _id: 2, video_name: "VIDEO2" },
  ],
  isLoading: false,
};

const SEARCH_RESULT = {
  data: [
    {
      confidence: "high",
      end: 25,
      score: 83.82,
      start: 20,
      video_id: "VIDEO1",
      author: "AUTHOR",
      filename: "VIDEO1",
    },
  ],
  isLoading: false,
};

const TEST_OPTIONS = {
  controls: true,
  responsive: true,
  fluid: true,
  sources: [],
  start: 20,
};

export { TEST_INDEXES, TEST_VIDEOS, SEARCH_RESULT, TEST_OPTIONS };
