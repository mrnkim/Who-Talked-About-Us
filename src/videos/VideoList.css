.filename-text {
    font-size: 0.8rem;
    margin: 0;
    line-height: 1;
}

.channelAndVideoName {
    display: flex;
    justify-content: flex-start;
    margin-top: 0.3rem;
    align-items: baseline;
  }

.filename-text {
    display: inline-flex;
    font-size: 0.75rem;
    padding: 0.375rem 0.375rem;
    align-items: flex-start;
    text-align: left;
  }

 .channelPillSmall {
    display: inline-flex;
    font-size: 0.75rem;
    line-height: 1rem;
    padding: 0.375rem 0.5rem;
    align-items: flex-start;
    border-radius: 2.5rem;
    background: var(--TL-Green-300, #C3F49C);
  }