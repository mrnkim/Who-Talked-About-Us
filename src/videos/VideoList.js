import { React, Suspense } from "react";
import { Col } from "react-bootstrap";
import ReactPlayer from "react-player";
import "./VideoList.css";
import LoadingSpinner from "../common/LoadingSpinner";
import { ErrorBoundary } from "react-error-boundary";
import keys from "../apiHooks/keys";
import ErrorFallback from "../common/ErrorFallback";

/** Shows list of the video in an index
 *
 *  VideoComponents -> VideoList
 *
 */

function VideoList({ videos, refetchVideos }) {
  const numVideos = videos.length;

  return videos.map((video, index) => (
    <ErrorBoundary
      FallbackComponent={ErrorFallback}
      onReset={() => refetchVideos()}
      resetKeys={[keys.VIDEOS, index]}
      key={video._id + "-" + index}
    >
      <Suspense fallback={<LoadingSpinner />}>
        <Col
          sm={12}
          md={numVideos > 1 ? 6 : 12}
          lg={numVideos > 1 ? 4 : 12}
          xl={numVideos > 1 ? 3 : 12}
          className="mb-5 mt-3"
        >
          {" "}
          <ReactPlayer
            url={video.metadata.youtubeUrl}
            controls
            width="100%"
            height="250px"
          />
          <div className="channelAndVideoName">
            <div className="channelPillSmall">{video.metadata.author}</div>
            <div className="filename-text">
              {video.metadata.filename.replace(".mp4", "")}
            </div>
          </div>
        </Col>
      </Suspense>
    </ErrorBoundary>
  ));
}

export default VideoList;
