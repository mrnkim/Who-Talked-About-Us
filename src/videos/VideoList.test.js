import React from "react";
import { render, screen } from "@testing-library/react";
import VideoList from "./VideoList.js";
import { TEST_VIDEOS } from "../common/_testCommon.js";

it("renders without crashing", function () {
  render(<VideoList videos={TEST_VIDEOS} />);
});

it("matches snapshot", function () {
  const { container } = render(<VideoList videos={TEST_VIDEOS} />);
  expect(container).toMatchSnapshot();
});

it("shows loading", function () {
  render(<VideoList videos={TEST_VIDEOS} />);
  const loadingElements = screen.getAllByText("Loading...");
  loadingElements.forEach((element) => {
    expect(element).toBeInTheDocument();
  });
});

//FIXME: fix test 
// it("shows delete button", async function () {
//   const deleteVideo = jest.fn(); // mock function
//   render(<VideoList videos={TEST_VIDEOS} deleteVideo={deleteVideo} />);
//   await waitFor(() => screen.getByText("Delete")); // wait for the 'Delete' button to appear
// });
