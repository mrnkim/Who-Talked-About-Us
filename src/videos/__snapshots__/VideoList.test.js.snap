// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="mb-4 col-xl-3 col-lg-4 col-md-6 col-sm-12"
  >
    <i>
      Loading...
    </i>
  </div>
  <div
    class="mb-4 col-xl-3 col-lg-4 col-md-6 col-sm-12"
  >
    <i>
      Loading...
    </i>
  </div>
</div>
`;
